src/main/java/bob/Parser.java

```diff
@@ -0,0 +1,298 @@
+                for (int i = 1; i < tasks.count; i++) {
+                    Task task = tasks.get(i);
+                    storage.appendToFile(filePath, task);
+                }
+            } catch (IOException e) {
+                System.out.println("Unable to write to file: " + e.getMessage());
+            }
+            System.out.println(mark + "   " + taskToMark.toString() + "\n" + line);
+            return;
+        case UNMARK:
+            // mark task as not done
+            System.out.println(line);
+            int indexToUnmark = Integer.valueOf(input.substring(7));
+            indexToUnmark--;
+            Task taskToUnmark = tasks.get(indexToUnmark);
+            taskToUnmark.markAsUndone();
+            try {
+                String filePath = "./data/tasks.txt";
+                Task firstTask = tasks.get(0);
+                storage.writeToFile(filePath, firstTask);
+                for (int i = 1; i < tasks.count; i++) {
+                    Task task = tasks.get(i);
+                    storage.appendToFile(filePath, task);
+                }
+            } catch (IOException e) {
+                System.out.println("Unable to write to file: " + e.getMessage());
+            }
+            System.out.println(unmark + "   " + taskToUnmark.toString() + "\n" + line);
+            return;
+        case DELETE:
+            // delete the task specified by the user
+            System.out.println(line + "\n" + delete);
+            int indexToDelete = Integer.valueOf(input.substring(7));
+            indexToDelete--;
+            Task taskToDelete = tasks.get(indexToDelete);
+            System.out.println(indent + " " + taskToDelete.toString());
+            tasks.remove(taskToDelete); // remove task from the list of tasks
+            tasks.count--; // decrement total count of tasks
+            try {
+                String filePath = "./data/tasks.txt";
+                File data = new File(filePath);
+                Task firstTask = tasks.get(0);
+                storage.writeToFile(filePath, firstTask);
+                for (int i = 1; i < tasks.count; i++) {
+                    Task task = tasks.get(i);
+                    if (storage.isNewFile) {
+                        storage.writeToFile(filePath, task);
+                        storage.isNewFile = false;
+                    } else {
+                        storage.appendToFile(filePath, task);
+                    }
+                }
+            } catch (IOException e) {
+                System.out.println("Unable to write to file: " + e.getMessage());
+            }
+            System.out.println(indent + "Now you have " + tasks.count + " tasks in the list.\n" + line);
+            return;
+        case CREATE:
+            // call helper method to create the task
+            Task task = createTask(input);
+            tasks.add(task);
+            try {
+                String filePath = "./data/tasks.txt";
```

Comment:
Any reason why in this file you opted to hardcode the file location instead of using the filePath from Bob.java?